webapp/bin/webapp-stop.js

```diff
@@ -36,41 +36,57 @@ function onTimeoutOccurred() {
  */
 function onConnectError(err) {
   logger.error("Could not connect due " + err.toString());
-  process.exit(1);
+  connectionAttempts++;
+  //Only finish the process by connection error, if cant connect in every socket (3 is for attempts)
+  if (connectionAttempts >= 3 * configQuantity ){
+    process.exit(1);
+  }
 }
 
+var configQuantity;
+var connectionAttempts = 0;
+
 try {
   Application.load();
 
-  var config = Application.getContextConfig();
-  var protocol = config.ssl ? "https://" : "http://";
-  var connURL = protocol + "0.0.0.0:" + config.port;
-
-  /**
-   * Socket options
-   * 
-   * @type {any}
-   */
-  var options = {
-    reconnectionAttempts: 3, // only three times
-    timeout: 10000 // 10 seconds
-  };
-  logger.info("Connecting to " + connURL);
+  var configList = Application.getAllConfigs();
+  var socketList = [];
+  for(key in configList){
+    var config = configList[key];
+    var protocol = config.ssl ? "https://" : "http://";
+    var connURL = protocol + "0.0.0.0:" + config.port;
+    /**
+     * Socket options
+     * 
+     * @type {any}
+     */
+    var options = {
+      reconnectionAttempts: 3, // only three times
+      timeout: 10000 // 10 seconds
+    };
+    logger.info("Connecting to " + connURL);
+  
+    socketList.push(require("socket.io-client")(connURL, options));
```

Comment:
Although nodejs requirer is like singleton mode, you should attach it into a variable and then call as functor.